L1Trigger/L1TTrackMatch/plugins/DisplacedVertexProducer.cc

```diff
@@ -0,0 +1,382 @@
+      if (dist_TPs(selectedTracksWithTruth[i].first, selectedTracksWithTruth[j].first) != 0)
+        continue;
+      Double_t x_dv_trk = -9999.0;
+      Double_t y_dv_trk = -9999.0;
+      Double_t z_dv_trk = -9999.0;
+      edm::Ptr<TrackingParticle> tp_i = selectedTracksWithTruth[i].second;
+      edm::Ptr<TrackingParticle> tp_j = selectedTracksWithTruth[j].second;
+      bool isReal = false;
+      if (!tp_i.isNull() && !tp_j.isNull()) {
+        bool isHard_i = false;
+        bool isHard_j = false;
+        if (!tp_i->genParticles().empty() && !tp_j->genParticles().empty()) {
+          isHard_i = tp_i->genParticles()[0]->isHardProcess() || tp_i->genParticles()[0]->fromHardProcessFinalState();
+          isHard_j = tp_j->genParticles()[0]->isHardProcess() || tp_j->genParticles()[0]->fromHardProcessFinalState();
+        }
+
+        if (tp_i->eventId().event() == 0 && tp_j->eventId().event() == 0 && fabs(tp_i->vx() - tp_j->vx()) < 0.0001 &&
+            fabs(tp_i->vy() - tp_j->vy()) < 0.0001 && fabs(tp_i->vz() - tp_j->vz()) < 0.0001 && isHard_i && isHard_j &&
+            ((tp_i->charge() + tp_j->charge()) == 0)) {
+          isReal = true;
+        }
+      }
+
+      int inTraj =
+          calcVertex(selectedTracksWithTruth[i].first, selectedTracksWithTruth[j].first, x_dv_trk, y_dv_trk, z_dv_trk);
+      Vertex_Parameters vertex = Vertex_Parameters(
+          x_dv_trk, y_dv_trk, z_dv_trk, selectedTracksWithTruth[i].first, selectedTracksWithTruth[j].first);
+
+      if (vertex.R_T > RTMax_)
+        continue;
+      if (vertex.R_T < RTMin_)
+        continue;
+
+      l1t::DisplacedTrackVertex outputVertex = l1t::DisplacedTrackVertex(selectedTracksWithTruth[i].first.index,
+                                                                         selectedTracksWithTruth[j].first.index,
+                                                                         inTraj,
+                                                                         vertex.d_T,
+                                                                         vertex.R_T,
+                                                                         vertex.cos_T,
+                                                                         vertex.delta_z,
+                                                                         vertex.x_dv,
+                                                                         vertex.y_dv,
+                                                                         vertex.z_dv,
+                                                                         vertex.openingAngle,
+                                                                         vertex.p_mag,
+                                                                         isReal);
+
+      if (runEmulation_) {
+        std::vector<ap_fixed<13, 8, AP_RND_CONV, AP_SAT>> Transformed_features = {
+            selectedTracksWithTruth[i].first.pt * 0.25,
```

Comment:
Can these magic number multiplications either be added to the python config or set as named variables before being used with an appropriate comment so that other users know what they are and what they are doing